.genx/utils.js

```diff
@@ -132,6 +134,27 @@ const parseJSONArgument = (name, defaultValue) =>
     }
   }
 
+const frameworkFolderMap = {
+  [ANGULAR]: 'angular',
+  [REACT]: 'react'
+}
+
+const excludeFrameworks = (selectedFramework) => {
+  const dir = path.join(__dirname, '..');
+  const ignoredFrameworks = FRAMEWORKS
+      .filter((framework) => framework !== selectedFramework)
+      .map((framework) => `client-${frameworkFolderMap[framework]}`);
+  ignoredFrameworks.forEach((framework) => {
+    const frameworkDirectory = `${dir}/${framework}`;
+    rmSync(frameworkDirectory, {recursive: true, force: true});
+  });
+  if (selectedFramework !== WEB_COMPONENTS) {
+    rmSync(`${dir}/client`, {recursive: true, force: true});
+    const frameworkDirectory = `${dir}/client-${frameworkFolderMap[selectedFramework]}`;
+    renameSync(frameworkDirectory, `${dir}/client`);
+  }
```

Comment:
This can be simplified quite a bit. We should also normalise framework name before processing (lowercase, replace spaces with hyphens), so angular/Angular/AnGular/Web Components/web-components/web components etc. would work equally.

Then you can do this:

```
const FRAMEWORKS = {
  ANGULAR: 'angular',
  WEB_COMPONENTS: 'web-components',
  REACT: 'react'
}
```

* we use the lower-case for the directory name, so we have client-tmp/angular, client-tmp/web-components, etc. I.e. we wouldn't need a separate `frameworkFolderMap`
* if we don't recognise framework name after normalisation, we can either print an error and exit or print an error and fallback to default - TBD
* logic in this file becomes really simple:
   * move client-tmp/${seletedFramework} to client
   * delete client-tmp

Reply:
my initial goal was to keep changes to minimum to avoid conflicts with rapid renaming but given that's merged it makes sense to follow your approach. @ArturKrasinski already implemented so let us know if it's ok

Reply:
Rapid stuff was only merged into `prerelease` branch so far. The plan is to merge it to `main` tomorrow together with the server (GSF 8) change.